pkg/cloud/libvirt/actuators/machine/actuator.go

```diff
@@ -408,7 +408,15 @@ func (a *Actuator) applyMachineStatus(
 	}
 
 	machineCopy := machine.DeepCopy()
-	machineCopy.Status.ProviderStatus = rawStatus
+
+	// Check if machine status has been modified.
+	machineStatus, err := ProviderStatusFromMachine(a.codec, machine)
+	if err != nil {
+		return false, err
+	}
+	if status != nil || !equality.Semantic.DeepEqual(machineStatus, status) {
+		machineCopy.Status.ProviderStatus = rawStatus
```

Comment:
Can you move `rawStatus, err := EncodeProviderStatus(a.codec, status)` to this block as this is the only place where this is used?